### PR TITLE
Switch to saner JSON dumping

### PR DESCRIPTION
**Pull request by mitsuhiko**
This is the correct fix for our JSON shenanigans.  This is also what Flask
does so we have prior experience there :)

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3803)

### CODE REVIEW

File: src/sentry/templatetags/sentry_api.py [hunk 1]
```diff
@@ -21,14 +20,12 @@ def serialize(context, obj):
...
-    return convert_to_json(serialize_func(obj, user), escape=True)
+    return convert_to_json(serialize_func(obj, user))
 
 
 @register.simple_tag
-def convert_to_json(obj, escape=False):
-    if escape:
-        return json.dumps_htmlsafe(obj)
-    return mark_safe(json.dumps(obj))
+def convert_to_json(obj):
```
Comment (mattrobenolt) [inline src/sentry/templatetags/sentry_api.py]: Do we not have cases where we don't want to escape JSON in templates? My hunch is "no", but I don't have anything to back that up.

File: src/sentry/utils/json.py [hunk 2]
```diff
@@ -70,6 +70,7 @@ def dump(value, fp, **kwargs):
 
 
 def dumps(value, escape=False, **kwargs):
+    # Legacy use. Do not use. Use dumps_htmlsafe
```
Comment (mattrobenolt) [inline src/sentry/utils/json.py]: What's wrong with this? Isn't it the same as `dumps_htmlsafe` except for the `mark_safe`?

File: src/sentry/templatetags/sentry_api.py (hunk 1 above)
Comment (mitsuhiko) [inline src/sentry/templatetags/sentry_api.py]: We should should stop using this method entirely. It's not used as a template tag from what i can tell. The `to_json` filter works now.

File: src/sentry/utils/json.py (hunk 2 above)
Comment (mitsuhiko) [inline src/sentry/utils/json.py]: It's bad naming. It's just html safe, it does not "escape" in the traditional sense. Also yes, it was not mark_safe.

**Comment by mattrobenolt** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3803)*

**Pull request merged**